packages/contracts/src/utils/erc721_wrapper.ts

```diff
@@ -0,0 +1,148 @@
+
+import { constants } from './constants';
+import { ContractName, ERC721BalancesByOwner } from './types';
+
+export class ERC721Wrapper {
+    private _tokenOwnerAddresses: string[];
+    private _contractOwnerAddress: string;
+    private _deployer: Deployer;
+    private _provider: Provider;
+    private _dummyERC721TokenContracts?: DummyERC721TokenContract[];
+    private _erc721ProxyContract?: ERC721ProxyContract;
+    private _initialERC721BalancesByOwner: ERC721BalancesByOwner = {};
+    constructor(deployer: Deployer, provider: Provider, tokenOwnerAddresses: string[], contractOwnerAddress: string) {
+        this._deployer = deployer;
+        this._provider = provider;
+        this._tokenOwnerAddresses = tokenOwnerAddresses;
+        this._contractOwnerAddress = contractOwnerAddress;
+    }
+    public async deployDummyERC721TokensAsync(): Promise<DummyERC721TokenContract[]> {
+        const tokenContractInstances = await Promise.all(
+            _.map(_.range(constants.NUM_DUMMY_ERC721_TO_DEPLOY), () =>
+                this._deployer.deployAsync(ContractName.DummyERC721Token, constants.DUMMY_ERC721_TOKEN_ARGS),
+            ),
+        );
+        this._dummyERC721TokenContracts = _.map(
+            tokenContractInstances,
+            tokenContractInstance =>
+                new DummyERC721TokenContract(tokenContractInstance.abi, tokenContractInstance.address, this._provider),
+        );
+        return this._dummyERC721TokenContracts;
+    }
+    public async deployERC721ProxyAsync(): Promise<ERC721ProxyContract> {
+        const proxyContractInstance = await this._deployer.deployAsync(ContractName.ERC721Proxy);
+        this._erc721ProxyContract = new ERC721ProxyContract(
+            proxyContractInstance.abi,
+            proxyContractInstance.address,
+            this._provider,
+        );
+        return this._erc721ProxyContract;
+    }
+    public async setBalancesAndAllowancesAsync() {
+        if (_.isUndefined(this._dummyERC721TokenContracts)) {
+            throw new Error('Dummy ERC721 tokens not yet deployed, please call "deployDummyERC721TokensAsync"');
+        }
+        if (_.isUndefined(this._erc721ProxyContract)) {
+            throw new Error('ERC721 proxy contract not yet deployed, please call "deployERC721ProxyAsync"');
+        }
+        const setBalancePromises: any[] = [];
+        const setAllowancePromises: any[] = [];
+        this._initialERC721BalancesByOwner = {};
+        _.forEach(this._dummyERC721TokenContracts, dummyERC721TokenContract => {
+            _.forEach(this._tokenOwnerAddresses, tokenOwnerAddress => {
+                _.forEach(_.range(constants.NUM_ERC721_TOKENS_TO_MINT), () => {
+                    const tokenId = ZeroEx.generatePseudoRandomSalt();
+                    setBalancePromises.push(
+                        dummyERC721TokenContract.mint.sendTransactionAsync(tokenOwnerAddress, tokenId, {
```

Comment:
What is the tokenId used for?

If it is supposed to be a replacement for hard-coded symbol/name for the token, can we make it deterministic based on index? Otherwise re-running the tests will generate new tokenIds and this will add confusion while debugging.

Reply:
I understand now, I think it should be ok.